File: src/com/github/shadowsocks/preferences/SummaryEditTextPreference.java
package com.github.shadowsocks.preferences;

import android.content.Context;
import android.preference.EditTextPreference;
import android.util.AttributeSet;

public class SummaryEditTextPreference extends EditTextPreference {

  private CharSequence mDefaultSummary;

  public SummaryEditTextPreference(Context context, AttributeSet attrs, int defStyle) {
    super(context, attrs, defStyle);
    mDefaultSummary = getSummary();
  }

  public SummaryEditTextPreference(Context context, AttributeSet attrs) {
    super(context, attrs);
    mDefaultSummary = getSummary();
  }

  public SummaryEditTextPreference(Context context) {
    super(context);
  }

  @Override
  public void setSummary(CharSequence summary) {
    if (summary.toString().isEmpty()) {
      super.setSummary(mDefaultSummary);
    } else {
      super.setSummary(summary);
    }
  }
}

File: src/main/scala/com/github/shadowsocks/preferences/SummaryEditTextPreference.scala
package com.github.shadowsocks.preferences

import android.content.Context
import android.preference.EditTextPreference
import android.util.AttributeSet

class SummaryEditTextPreference(context: Context, attrs: AttributeSet, defStyle: Int)
  extends EditTextPreference(context, attrs, defStyle) {

  def this(context: Context, attrs: AttributeSet) = {
    this(context, attrs, android.R.attr.editTextPreferenceStyle)
    mDefaultSummary = getSummary
  }

  override def setSummary(summary: CharSequence) {
    if (summary.toString.isEmpty) {
      super.setSummary(mDefaultSummary)
    } else {
      super.setSummary(summary)
    }
  }

  private var mDefaultSummary: CharSequence = getSummary
}